### PR TITLE
Trying to correct the documentation so that it actually describes how the code behaves. (Attempt two)

### PR DESCRIPTION
**Pull request by lemire**
Documentation update.

I think that casts and get<type>() are effectively equivalent which brings up the question: why do we bother with  get<type>() at all when exceptions are enabled?

Proof...

```C++

inline simdjson_result<dom::element>::operator bool() const noexcept(false) {
  return get<bool>();
}
inline simdjson_result<dom::element>::operator const char *() const noexcept(false) {
  return get<const char *>();
}
inline simdjson_result<dom::element>::operator std::string_view() const noexcept(false) {
  return get<std::string_view>();
}
inline simdjson_result<dom::element>::operator uint64_t() const noexcept(false) {
  return get<uint64_t>();
}
inline simdjson_result<dom::element>::operator int64_t() const noexcept(false) {
  return get<int64_t>();
}
inline simdjson_result<dom::element>::operator double() const noexcept(false) {
  return get<double>();
}
inline simdjson_result<dom::element>::operator dom::array() const noexcept(false) {
  return get<dom::array>();
}
inline simdjson_result<dom::element>::operator dom::object() const noexcept(false) {
  return get<dom::object>();
}
```

See issue https://github.com/simdjson/simdjson/issues/711

This is the second attempt (previously the fuzzer were busted).

Fixes https://github.com/simdjson/simdjson/issues/711

### CODE REVIEW

**Comment by lemire** [comment]
@pauldreik Can you help? It looks like the fuzzers are busted and it is not clear to me how to fix it. It says something about an internal error... I do not know how to fix this.

**Comment by lemire** [comment]
Here is the error we encounter:

"GitHub Actions has encountered an internal error when running your job."

**Comment by lemire** [comment]
Trying again to see if the github actions will stop erroring.

**Comment by lemire** [comment]
@jkeiser I have tweaked the little paragraph so it is as nice as possible.

File: doc/basics.md [hunk 1]
```diff
@@ -57,9 +57,20 @@ Once you have an element, you can navigate it with idiomatic C++ iterators, oper
...
+    double value;
+    result.get<double>().tie(value, error); // no exception, ever
+    /**
+     * If you do not have to support libc++, you can use the simpler syntax:
+     * auto [value, error] = result.get<double>();
+     */
+    if(error) {
+      // error case
+    }
+    ```
```
Comment (jkeiser) [inline doc/basics.md:73]: The code and trailing \`\`\` seems to be misaligned from the \`\`\`c++, I don't know if that's going to render right on all markdown viewers ...

Comment (lemire) [reply doc/basics.md]: Fixed

**Comment by lemire** [comment]
> We need to add code snippets to readme_examples*.cpp if we add it here. Just to keep ourselves 100% honest that things actually compile.

Done. Of course, the end result is that we need a full working example and it is much longer.

**Comment by pauldreik** [comment]
> @pauldreik Can you help? It looks like the fuzzers are busted and it is not clear to me how to fix it. It says something about an internal error... I do not know how to fix this.

I saw the weird job result, seems to have been some glitch. Later jobs have worked, and I could rerun the same job (still running, seems to work fine). I guess github actions must be absolutely bullet proof since they accept public input, so I do not think anything the job could do would not be detected and seen in the log!

**Comment by jkeiser** [comment]
> Done. Of course, the end result is that we need a full working example and it is much longer.

I compromised here by allowing a nonessential variable such as filename to be specified outside the example, and when I pasted the examples into functions, I set the variables up above (but didn't touch the function).

**Comment by lemire** [comment]
@jkeiser I could truncate the working example if you think that's better. I have no specific opinion on the matter. 

That's not what is important in this PR: this PR fixes an out of sync documentation which is likely to throw off users.

Let me know what you think. I would argue that cosmetic changes can be fixed with other commits. One can even reword things further... The essential point is to fix the bad documentation.

**Comment by lemire** [comment]
Ok. I checked in a slightly smaller piece of code. Let us see if the tests are happy.

**Comment by jkeiser** [comment]
@lemire thanks for bearing with me there. I'd hate for examples to be such a pain in the ass to add that we don't write them, so I'm hoping verbatim copy/paste of snippets is a reasonable compromise.

**Comment by lemire** [comment]
Merging.

**Pull request merged**